### ISSUE TITLE
Quick Start Mode

### ISSUE THREAD

**Issue by kiranetph** [Serhioromano/cobalt-9#42]
How cool would it be to have a "Quick Start" mode, where we only type the SECTION name into a form and the script will create the **MENU ITEM** for it then a **SECTION** (with default parameters) at the same time the **TYPE** (link it to the **SECTION**) and assign the `menu item id` to **SECTION** and **TYPE**.

A nice addition would be an **optional** field ("Categories"), where you can enter a list (maybe comma separated) of first level categories, which will be automatically generated under the section.

A third **optional** field could hold a list of field names, which will be added (maybe just **text** fields as default).

Then we can start fine-tuning the **SECTION**, **TYPE** and **TEMPLATE** parameters.

![2016-01-20_17-22-46](https://cloud.githubusercontent.com/assets/15226809/12445396/fde11534-bf9e-11e5-9a0a-77f426afce76.png)

**Comment by Serhioromano** [comment]
This is nice idea. It is like application creation wizard. I can add select field to select in what menu create element and enter name of the menu, maybe you need few types, and express settings like do you want personalization, rating, tags, ....

**Comment by kiranetph** [comment]
That would be already the next level or a sort of  "advanced" mode. 
The main target for this is a new user, who can get something up and running in typing only 1 **SECTION** name (or some additional field names). Of course it will also help an experienced user as well. 
RAD (Rapid Application Development) at is best ... 

This would be a "killer feature" and can be enhanced along the way (predefined settings like a template) etc...

**Comment by andiamine** [comment]
Yes i agree too, quick start mode will be very good !

**Comment by pepperstreet** [comment]
I am always a fan of simplicity and better UX, 
and I agree on the "annoying back and forth" in the initial creation of a Cobalt "app"... 

But isn't all this available by the **Packs** feature !? 
I would suggest to supply very **simple starter packs** as additional **free downloads**.
This allows a simple 1-click upload and installation for everyone. Even for the novice user or "DAU" ;)

**Comment by kiranetph** [comment]
The packs provide a certain ready-made solution and might not always fit the initial setup (no special fields). Packs are actually general solutions while  "Quick Start" can be seen as a personalised starter pack without record data.
"Quick Start" would be very helpful for experienced users as well.
Just imagine you want to create a **SECTION/TYPE** with 10 categories --> all you need to do is type the **SECTION** name into an entry field and all 10 categories separated by comma into another, then press submit -- That's it ... 
With a "Starter Pack" more adjustments are needed to do -> rename default **TYPE**, rename default **SECTION**, create or rename categories, create menu link ....

**Comment by Serhioromano** [comment]
the nice thing here you do not need to go to backend to create menu element. I am on it. I think this is very important feature.

**Comment by kiranetph** [comment]
This will change the usability to a new level.
Another headache is the creation of a parent-child setup. Sometimes people are confused by the settings.
Will add later more detailed example how to address it nicely with a "quick start" module, or better said with an enhanced mode.

**Comment by justanotherguyme** [comment]
I agree. I use cobalt everyday and I still have to watch the video on setting it up. This is a must have.

**Comment by Serhioromano** [comment]
> and I agree on the "annoying back and forth" in the initial creation of a Cobalt "app"...

This wizard will not obsolete that type of work. If the wizard will have minimal settings, the user will still have to go back and forth.

But you are right @pepperstreet  pack is already sort of quick start.

This idea is nice but there are so many questions. Will those be different forms for types and sections? Or the same? If the same it will show the same parameters? Or it will be one form whee you enter section name, type and list of fields? But what about parameters? Which one is important and which one is not? 

I would close this for now because I do not see this as realistic at this stage. I have to distribute my work and chose what to do and what to skip.

**Comment by kiranetph** [comment]
I see it different from a pack. 
It was always the start of any Section/Type setup (Incl the itemID) which created the biggest hurdle to take when working with Cobalt. 
If this can be taken away from beginners, it will definitely attribute to a better user satisfaction. 
Don't close the issue for the moment. I will check if I still have my old notes. I remember that I wrote down quite some ideas, also regarding default settings for a quick start module.

**Comment by kiranetph** [comment]
Especially now with the move to frontend editing, it is even more frustrating to create a working SECTION/TYPE set. Just think of newcomers and the expected level of frustration.

New workflow under Cobalt 9:
1) Create a section in the frontend
2) log in to backend (admin) and create a menu item
3) change back to frontend, select section and adjust Item ID
4) create TYPE associated with section and respective item ID 

New (very basic) Quick-Start Feature: 
1) Enter just the Name of a section and submit the form, then Cobalt9 will
a) Create a section with all default parameter
b) Create the respective menu item 
c) update section with item ID
d) create TYPE with same name, default parameters incl. item ID

To fine-tune the process, optional parameters (categories, fields, etc) can be introduced later along the way.  

There are almost no PACKS available, but the Quickstart will enable EVERY user to overcome the first hurdle when starting with Cobalt. --> creating a working environment without problems and confusing "back and forth" steps.

PACKS represent a specialised environment/solution. 
Quickstart will simplify a mandatory process.

**Comment by Serhioromano** [comment]
I home my day was not lost. Although I do not see like it is very needed, I have created it. I've been proved to be wrong on so many points. Perhaps those are things that keep Cobalt from popularity. So I'll listen to you although I personally do not see it as critical, but understand that my certainty my be way too contorted. 

So start with CP

![2019-08-28_20-32-35](https://user-images.githubusercontent.com/650741/63865437-687fd000-c9d3-11e9-85d1-3e27a4eb8dec.jpg)

Then simple form
 
![2019-08-28_20-32-53](https://user-images.githubusercontent.com/650741/63865464-72a1ce80-c9d3-11e9-94a6-09009a5a473d.jpg)

I'll create a section, type, 2 fields, menu link, bind all together, create section description with links what to do next.

**Comment by Serhioromano** [comment]
I'll release an update in a few minutes.

**Comment by kiranetph** [comment]
it is not critical, but it will tremendously help beginners ... just think of the confusion before regarding item ID and the resulting mistakes .. also the back and forth .. now you just type the section name and all the "headache" from before is gone and everything is provided magically ... I will test it ...

**Comment by Serhioromano** [comment]
>  I will test it ...

I'll be waiting for your response.

**Comment by kiranetph** [comment]
Wahhhh ... how cool is this!!!!!!!

Remember, before a lot of beginner did not mark the TYPE under submit type parameter in SECTION. This all is now taken care of. Before you need minimum of 5 minutes to create everything, now 5 sec. !!!!!

Missing is still  the correct Item/menu ID under TYPE.
Not yet sure about the 2 additional fields. Have to think about it. 
I would make the TYPE entry optional and not mandatory. 
If nothing is typed in, the same name as the section will be used. 

With a little more optional fine-tuning for some essential parameter, this is something which will make Cobalt9 extremely easy for starters. !!!!!!!!!

**Comment by Serhioromano** [comment]
> Missing is still the correct Item/menu ID under TYPE.

Type menu ID inherits from a section. It is needed if only you want different menu ID for article list and full article views.

> Not yet sure about the 2 additional fields. Have to think about it.

Me too. It is easy to delete. But what I wanted is to show a list of fields and users may always delete those fields. I am not sure maybe it is not needed extra.

**Comment by kiranetph** [comment]
The created menu link can not be edited from Joomla backend
![image](https://user-images.githubusercontent.com/15226809/63913470-d7f0d080-ca62-11e9-8842-ec360f5223e2.png)

**Comment by kiranetph** [comment]
> It is easy to delete. But what I wanted is to show a list of fields and users may always delete those fields. I am not sure maybe it is not needed extra.

Lets see later.

I would suggest to make the Type name optional.
If no TYPE name is encoded, it will take the section name.

For the menu link, I was checking what is a eventually needed from the start. I believe you should add an ACCESS dropdown to choose the access group and an optional menu title. The menu title would be optional. If empty it would use the SECTION name.
--> Menu link would be 3 fields:
1) Title (optional) : if empty take section title
2) Menu Group : default is Main Menu
3) Access Group: default is Public

**Comment by Serhioromano** [comment]
> The created menu link can not be edited from Joomla backend
> ![image](https://user-images.githubusercontent.com/15226809/63913470-d7f0d080-ca62-11e9-8842-ec360f5223e2.png)

It should be fixed if you download the latest media pack and component and update.

**Comment by kiranetph** [comment]
> It should be fixed if you download the latest media pack and component and update.

I have Media Pack 9.33 and Cobalt 9.71 and it does not work
